### PR TITLE
Memory handling fixes

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/windows/editor_object_selection.c [hunk 1]
```diff
@@ -354,6 +354,10 @@ static void visible_list_refresh(rct_window *w)
 	case RIDE_SORT_RIDE:
 		sortFunc = visible_list_sort_ride_name;
 		break;
+	default:
+		log_error("Wrong sort type %d, leaving list as-is.", _listSortType);
```
Comment (IntelOrca) [inline src/windows/editor_object_selection.c]: not an error, use log_warning

File: src/windows/loadsave.c [hunk 2]
```diff
@@ -602,6 +602,7 @@ static void window_loadsave_populate_list(int includeNewItem, bool browsable, co
 	strncpy(_directory, directory, sizeof(_directory));
 	if (_extension != extension) {
 		strncpy(_extension, extension, sizeof(_extension));
+		_extension[sizeof(_extension) - 1] = '\0';
```
Comment (IntelOrca) [inline src/windows/loadsave.c]: I keep suggesting you make a helper function which does a strncpy + null terminator for you.

File: src/windows/editor_scenario_options.c [hunk 3]
```diff
@@ -559,7 +559,7 @@ static void window_editor_scenario_options_financial_mousedown(int widgetIndex,
 		break;
 	}
 
-	if(RCT2_GLOBAL(RCT2_ADDRESS_SCREEN_FLAGS, uint8) & SCREEN_FLAGS_PLAYING) {
+	if(RCT2_GLOBAL(RCT2_ADDRESS_SCREEN_FLAGS, uint8) == SCREEN_FLAGS_PLAYING) {
 		window_invalidate_by_class(WC_FINANCES);
```
Comment (duncanspumpkin) [inline src/windows/editor_scenario_options.c]: I have a strong feeling this should stay as and. I will double check tomorrow.

Comment (IntelOrca) [inline src/windows/editor_scenario_options.c]: This was a mistake made in 0491d971f3ad79e6327b3fc14d8180241ca5a107. This is right as it should only invalidate the finance related windows if you are changing things in this window in-game.

See:
https://github.com/janisozaur/OpenRCT2/commit/0491d971f3ad79e6327b3fc14d8180241ca5a107#diff-31802840e0eb7d8cc0b1347e0883b20bR569

Comment (janisozaur) [inline src/windows/editor_scenario_options.c]: Should this go in then or should I remove that block of code (as it's dead code right now)?

Comment (duncanspumpkin) [inline src/windows/editor_scenario_options.c]: Yeah sounds like it's fine

**Comment by janisozaur** [comment]
Can this be merged now then? I'll try to make that function next time, @IntelOrca.

**Pull request merged**